src/processing/processing.ts

```diff
@@ -158,3 +167,129 @@ export const postProcessApiCallResponse = async (
 
   return goResult.data;
 };
+
+/**
+ * Pre-processes API call parameters based on the provided endpoint's processing specifications.
+ *
+ * @param endpoint The endpoint containing processing specifications.
+ * @param apiCallParameters The parameters to be pre-processed.
+ * @param processingOptions Options to control the async processing behavior like retries and timeouts.
+ *
+ * @returns A promise that resolves to the pre-processed parameters.
+ */
+export const preProcessApiCallParametersV2 = async (
+  endpoint: Endpoint,
+  apiCallParameters: ApiCallParameters,
+  processingOptions: GoAsyncOptions = { retries: 0, totalTimeoutMs: DEFAULT_PROCESSING_TIMEOUT_MS }
+): Promise<PreProcessingV2Response> => {
+  const { preProcessingSpecificationV2 } = endpoint;
+  if (!preProcessingSpecificationV2) return { apiCallParameters };
+
+  // We only wrap the code through "go" utils because of the timeout and retry logic.  In case of error, the function
+  // just re-throws.
+  const goProcessedParameters = await go(async () => {
+    const { environment, timeoutMs, value } = preProcessingSpecificationV2;
+
+    switch (environment) {
+      case 'Node': {
```

Comment:
Just the `Node` case here?

Reply:
Yeah. I documented it here https://github.com/api3dao/vitepress-docs/pull/669/files#diff-a6763a0e3838650f30686b3ab53e68be092b87ae305ec9e576fe6b6153e54b09R116

basically the new processing works with both sync and async code so it's better to just use `Node`.

Reply:
Do we need an `environment` at all then if there's only 1 possible value? Future proofing?

Reply:
Yeah, for example [Node permissioned](https://nodejs.org/api/permissions.html#:~:text=1%20%2D%20Active%20development-,The%20Node.,the%20%2D%2Dexperimental%2Dpermission%20flag.)